### PR TITLE
HDDS-3672. Ozone fs failed to list intermediate directory.

### PR DESCRIPTION
**Pull request by ChenSammi**
## What changes were proposed in this pull request?

The root cause is replication factor and replication type are required fieldsin KeyInfo proto. 

While the way to create the OMKeyInfo object for intermediate directory doesn't provide all required fields.  

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3672

## How was this patch tested?

Unit test is updated + manual confirmation:

```
bin/ozone sh volume create /hadoop
bin/ozone sh bucket create /hadoop/ozone
bin/ozone sh key put /hadoop/ozone/benchmarks/TestDFSIO/dir1/test README.md
bin/ozone fs -ls o3fs://ozone.hadoop/
```

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java [hunk 1]
```diff
@@ -465,11 +464,11 @@ public static ObjectName registerWithJmxProperties(
...
   /**
-   * Get the current UTC time in milliseconds.
-   * @return the current UTC time in milliseconds.
+   * Get the current time in milliseconds.
+   * @return the current time in milliseconds.
    */
-  public static long getUtcTime() {
-    return Calendar.getInstance(UTC_ZONE).getTimeInMillis();
+  public static long getTime() {
+    return System.currentTimeMillis();
```
Comment (rakeshadr) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java:471]: Thanks @ChenSammi for the analysis and the changes.

It looks like the "current time" change is not related to this, please let me know your thoughts.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java [hunk 2]
```diff
@@ -2037,11 +2037,11 @@ private void listStatusFindKeyInTableCache(
                 // if entry is a directory
                 if (!deletedKeySet.contains(entryInDb)) {
                   if (!entryKeyName.equals(immediateChild)) {
-                    OmKeyInfo fakeDirEntry = new OmKeyInfo.Builder()
-                        .setVolumeName(omKeyInfo.getVolumeName())
-                        .setBucketName(omKeyInfo.getBucketName())
-                        .setKeyName(immediateChild)
-                        .build();
+                    OmKeyInfo fakeDirEntry = createDirectoryKey(
```
Comment (rakeshadr) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java:2040]: It looks to me an opportunity to add a unit test, can we? Thanks!

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java (hunk 1 above)
Comment (ChenSammi) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java:471]: It's part of another PR.  Will upload a clean patch.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java (hunk 2 above)
Comment (ChenSammi) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java:2040]: Sure.

Comment (ChenSammi) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java:2040]: I find an easy way to reproduce the issue.  Use "key put" to upload an object with a path like object name, for example "object-dir/object-name", then use "fs -ls" to list the bucket. 
Will try to add a UT too.

**Comment by ChenSammi** [comment]
Thanks @elek  and @rakeshadr  for the review.
